File: 9. Palindrome Number/9. Palindrome Number.py
#
# Problem: 9. Palindrome Number
# Difficulty: Easy
# Link: https://leetcode.com/problems/palindrome-number/
# Language: python
# Date: 2026-04-18


class Solution(object):
    def isPalindrome(self, x):
        if x<0 :
            return False
        temp = x
        rev = 0
        while(temp != 0):
            t = temp%10
            rev = (rev*10)+t
            temp = temp/10

        if rev==x:
            return True
        return False    
        

File: 9. Palindrome Number/9. Palindrome Number.cpp
/*
 * Problem: 9. Palindrome Number
 * Difficulty: Easy
 * Link: https://leetcode.com/problems/palindrome-number/
 * Language: cpp
 * Date: 2026-04-18
 */

class Solution {
public:
    bool isPalindrome(int x) {

        // Store original value to compare later
        int temp = x;

        // long long to escape from overflow
        long long rev = 0;

        // Negative numbers are never palindrome
        if (x < 0) return false;

        // Reverse the number
        while (temp != 0) {

            // Get last digit
            int t = temp % 10;

            // Add digit to reversed number
            rev = (rev * 10) + t;

            // Remove last digit from temp
            temp = temp / 10;
        }

        // Check if original number and reversed number are same
        if (rev == x) return true;

        // If not same, it's not a palindrome
        return false;
    }
};
